src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1458,23 +1448,16 @@ private static void SortAndRemoveDuplicates<T>(List<T> input, Func<T, string> ke
                 }
             );
 
-            bool firstItem = true;
             string previousKey = null;
-            List<T> output = new List<T>(input.Count);
-            foreach (T item in input)
+            input.RemoveAll(ShouldRemove);
+
+            bool ShouldRemove(T item)
             {
                 string currentKey = keyGetter(item);
-                if ((firstItem) || !currentKey.Equals(previousKey, StringComparison.OrdinalIgnoreCase))
-                {
-                    output.Add(item);
-                }
-
+                bool match = previousKey != null && currentKey.Equals(previousKey, StringComparison.OrdinalIgnoreCase);
```

Comment:
What if item-s is null? Second null will be not removed?

Reply:
This was my thought, but this is private code, and `keyGetter` never returns null in practice.

I think it would be best to add null annotations and maybe a `Debug.Assert`.

Reply:
> I think it would be best to add null annotations

We're doing it at such a rate that it will take us a hundred years to finish the job. 😸 
Adding `#nullable enable` locally add no value and  doesn't protect from errors coming from highest level of code.
`Debug.Assert` assumes we cover the code fully by tests but we don't run tests on Debug build at all.

Nevertheless I'd agree with local `#nullable enable` if we add a comment that sessionState.ExportedFunctions, sessionState.Module.CompiledExports, sessionState.ExportedVariables and sessionState.ExportedAliases do not contain nulls by design.